from PyQt6 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(800, 210)
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(14)
        MainWindow.setFont(font)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("./Vid.ico"), QtGui.QIcon.Mode.Normal, QtGui.QIcon.State.Off)
        MainWindow.setWindowIcon(icon)
        MainWindow.setStyleSheet("color:rgb(180, 180, 180);\n"
"background-color: rgb(42, 42, 42);\n")
        MainWindow.setIconSize(QtCore.QSize(32, 32))
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(20)
        self.centralwidget.setFont(font)
        self.centralwidget.setObjectName("centralwidget")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(0, 0, 800, 40))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(20)
        self.label.setFont(font)
        self.label.setStyleSheet("color:rgb(255, 255, 255);")
        self.label.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.label.setObjectName("label")
        self.groupBox = QtWidgets.QGroupBox(self.centralwidget)
        self.groupBox.setGeometry(QtCore.QRect(10, 40, 186, 100))
        self.groupBox.setStyleSheet("font: 14pt \"Tahoma\";\n"
"border: 0px solid;")
        
        self.groupBox.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.groupBox.setObjectName("groupBox")
        self.widget = QtWidgets.QWidget(self.groupBox)
        self.widget.setGeometry(QtCore.QRect(10, 20, 171, 71))
        self.widget.setObjectName("widget")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.widget)
        self.verticalLayout.setContentsMargins(10, 0, 0, 0)
        self.verticalLayout.setSpacing(10)
        self.verticalLayout.setObjectName("verticalLayout")
        self.radioButton = QtWidgets.QRadioButton(self.widget)
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(12)
        font.setBold(False)
        font.setItalic(False)
        self.radioButton.setFont(font)
        self.radioButton.setChecked(True)
        self.radioButton.setStyleSheet("font: 12pt \"Tahoma\";")
        self.radioButton.setObjectName("radioButton")
        self.verticalLayout.addWidget(self.radioButton)
        self.radioButton_2 = QtWidgets.QRadioButton(self.widget)
        self.radioButton_2.setStyleSheet("font: 12pt \"Tahoma\";")
        self.radioButton_2.setObjectName("radioButton_2")
        self.verticalLayout.addWidget(self.radioButton_2)
        self.groupBox_2 = QtWidgets.QGroupBox(self.centralwidget)
        self.groupBox_2.setGeometry(QtCore.QRect(208, 40, 186, 100))
        self.groupBox_2.setStyleSheet("font: 14pt \"Tahoma\";\n"
"border: 0px solid;")
        self.groupBox_2.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.groupBox_2.setObjectName("groupBox_2")
        self.spinBox = QtWidgets.QSpinBox(self.groupBox_2)
        self.spinBox.setGeometry(QtCore.QRect(20, 30, 50, 20))
        self.spinBox.setStyleSheet("font: 12pt \"Tahoma\";\n"
"background-color: rgb(255, 255, 255);\n"
"color:rgb(0, 0, 0);\n"
"")
        self.spinBox.setButtonSymbols(QtWidgets.QAbstractSpinBox.ButtonSymbols.PlusMinus)
        self.spinBox.setSuffix("")
        self.spinBox.setPrefix("")
        self.spinBox.setMaximum(51)
        self.spinBox.setObjectName("spinBox")
        self.groupBox_3 = QtWidgets.QGroupBox(self.centralwidget)
        self.groupBox_3.setGeometry(QtCore.QRect(604, 40, 186, 100))
        self.groupBox_3.setStyleSheet("font: 14pt \"Tahoma\";\n"
"border: 0px solid;")
        self.groupBox_3.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.groupBox_3.setObjectName("groupBox_3")
        self.comboBox_2 = QtWidgets.QComboBox(self.groupBox_3)
        self.comboBox_2.setGeometry(QtCore.QRect(20, 30, 146, 20))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(12)
        font.setBold(False)
        font.setItalic(False)
        self.comboBox_2.setFont(font)
        self.comboBox_2.setStyleSheet("selection-color: rgb(255, 255, 255);\n"
"font: 12pt \"Tahoma\";background-color: rgb(50, 50, 50);")
        self.comboBox_2.setCurrentText("")
        self.comboBox_2.setInsertPolicy(QtWidgets.QComboBox.InsertPolicy.NoInsert)
        self.comboBox_2.setObjectName("comboBox_2")
        self.comboBox_2.addItem("")
        self.comboBox_2.addItem("")
        self.comboBox_2.addItem("")
        self.comboBox_2.addItem("")
        self.comboBox_2.addItem("")
        self.comboBox_2.addItem("")
        self.groupBox_4 = QtWidgets.QGroupBox(self.centralwidget)
        self.groupBox_4.setGeometry(QtCore.QRect(406, 40, 186, 100))
        self.groupBox_4.setStyleSheet("font: 14pt \"Tahoma\";\n"
"border: 0px solid;")
        self.groupBox_4.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.groupBox_4.setObjectName("groupBox_4")
        self.comboBox = QtWidgets.QComboBox(self.groupBox_4)
        self.comboBox.setGeometry(QtCore.QRect(20, 30, 146, 20))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(12)
        font.setBold(False)
        font.setItalic(False)
        self.comboBox.setFont(font)
        self.comboBox.setStyleSheet("selection-color: rgb(255, 255, 255);\n"
"font: 12pt \"Tahoma\";background-color: rgb(50, 50, 50);")
        self.comboBox.setCurrentText("")
        self.comboBox.setInsertPolicy(QtWidgets.QComboBox.InsertPolicy.NoInsert)
        self.comboBox.setObjectName("comboBox")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton.setGeometry(QtCore.QRect(360, 140, 130, 25))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(14)
        self.pushButton.setFont(font)
        self.pushButton.setObjectName("pushButton")
        self.label.raise_()
        self.groupBox.raise_()
        self.groupBox_2.raise_()
        self.groupBox_4.raise_()
        self.groupBox_3.raise_()
        self.pushButton.raise_()
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 800, 18))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(12)
        font.setBold(False)
        font.setItalic(False)
        self.menubar.setFont(font)
        self.menubar.setStyleSheet("font: 10pt \"Tahoma\";\n"
"selection-background-color: rgb(0, 0, 0);\n"
"background-color: rgb(56, 56, 56)")
        self.menubar.setObjectName("menubar")
        self.menuChoose_file = QtWidgets.QMenu(self.menubar)
        self.menuChoose_file.setToolTipsVisible(True)
        self.menuChoose_file.setObjectName("menuChoose_file")
        self.menuSettings = QtWidgets.QMenu(self.menubar)
        self.menuSettings.setObjectName("menuSettings")
        self.menuHelp = QtWidgets.QMenu(self.menubar)
        self.menuHelp.setObjectName("menuHelp")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setSizeGripEnabled(True)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)
        self.actionNew = QtGui.QAction(MainWindow)
        self.actionNew.setObjectName("actionNew")

        self.actionHelp = QtGui.QAction(MainWindow)
        self.actionHelp.setObjectName("actionHelp")
        self.actionSettings = QtGui.QAction(MainWindow)
        self.actionSettings.setObjectName("actionSettings")
        
        self.menuChoose_file.addAction(self.actionNew)
        self.menuSettings.addAction(self.actionSettings)
        self.menuHelp.addAction(self.actionHelp)
        self.menubar.addAction(self.menuChoose_file.menuAction())
        self.menubar.addAction(self.menuSettings.menuAction())
        self.menubar.addAction(self.menuHelp.menuAction())
        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)
     

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(QtCore.QCoreApplication.translate("MainWindow", "Ultimate Video Converter"))
        self.label.setText(QtCore.QCoreApplication.translate("MainWindow", "Ultimate Video Converter"))
        self.groupBox.setTitle(QtCore.QCoreApplication.translate("MainWindow", "Encoding Codec"))
        self.radioButton.setToolTip(QtCore.QCoreApplication.translate("MainWindow", "<html><head/><body><p>H264</p></body></html>"))
        self.radioButton.setText(QtCore.QCoreApplication.translate("MainWindow", "Default"))
        self.radioButton_2.setToolTip(QtCore.QCoreApplication.translate("MainWindow", "<html><head/><body><p><span style=\" font-size:10pt;\">H265</span></p></body></html>"))
        self.radioButton_2.setText(QtCore.QCoreApplication.translate("MainWindow", "HEVC"))
        self.groupBox_2.setTitle(QtCore.QCoreApplication.translate("MainWindow", "Video Quality"))
        self.spinBox.setToolTip(QtCore.QCoreApplication.translate("MainWindow", "<html><head/><body><p><span style=\" font-size:10pt;\">From 0 to 51. </span></p><p><span style=\" font-size:10pt;\">(0 - best quality, 51 - the worst quality)</span></p></body></html>"))
        self.groupBox_3.setTitle(QtCore.QCoreApplication.translate("MainWindow", "Video Tune"))
        self.comboBox_2.setToolTip(QtCore.QCoreApplication.translate("MainWindow", "<html><head/><body><p><span style=\" font-size:10pt;\">Choose Video Tune</span></p></body></html>"))
        self.comboBox_2.setPlaceholderText(QtCore.QCoreApplication.translate("MainWindow", "None"))
        self.comboBox.setItemText(0, QtCore.QCoreApplication.translate("MainWindow", "Ultra Fast"))
        self.comboBox.setItemText(1, QtCore.QCoreApplication.translate("MainWindow", "Super Fast"))
        self.comboBox.setItemText(2, QtCore.QCoreApplication.translate("MainWindow", "Very Fast"))
        self.comboBox.setItemText(3, QtCore.QCoreApplication.translate("MainWindow", "Faster"))
        self.comboBox.setItemText(4, QtCore.QCoreApplication.translate("MainWindow", "Fast"))
        self.comboBox.setItemText(5, QtCore.QCoreApplication.translate("MainWindow", "Medium"))
        self.comboBox.setItemText(6, QtCore.QCoreApplication.translate("MainWindow", "Slow"))
        self.comboBox.setItemText(7, QtCore.QCoreApplication.translate("MainWindow", "Slower"))
        self.comboBox.setItemText(8, QtCore.QCoreApplication.translate("MainWindow", "Very Slow"))
        self.groupBox_4.setTitle(QtCore.QCoreApplication.translate("MainWindow", "Conversion Speed"))
        self.comboBox.setToolTip(QtCore.QCoreApplication.translate("MainWindow", "<html><head/><body><p><span style=\" font-size:10pt;\">The slower the conversion </span></p><p><span style=\" font-size:10pt;\">the better the quality</span></p></body></html>"))
        self.comboBox.setPlaceholderText(QtCore.QCoreApplication.translate("MainWindow", "None"))
        self.comboBox_2.setItemText(0, QtCore.QCoreApplication.translate("MainWindow", "Film"))
        self.comboBox_2.setItemText(1, QtCore.QCoreApplication.translate("MainWindow", "Animation"))
        self.comboBox_2.setItemText(2, QtCore.QCoreApplication.translate("MainWindow", "Grain"))
        self.comboBox_2.setItemText(3, QtCore.QCoreApplication.translate("MainWindow", "Still Image"))
        self.comboBox_2.setItemText(4, QtCore.QCoreApplication.translate("MainWindow", "Fast Decode"))
        self.comboBox_2.setItemText(5, QtCore.QCoreApplication.translate("MainWindow", "Zero Latency"))
        self.pushButton.setToolTip(QtCore.QCoreApplication.translate("MainWindow", "<html><head/><body><p><span style=\" font-size:10pt;\">Start Video Conversion</span></p></body></html>"))
        self.pushButton.setText(QtCore.QCoreApplication.translate("MainWindow", "Convert"))
        
        self.menuChoose_file.setTitle(QtCore.QCoreApplication.translate("MainWindow", "File"))
        self.menuHelp.setTitle(QtCore.QCoreApplication.translate("MainWindow", "Help"))
        self.menuSettings.setTitle(QtCore.QCoreApplication.translate("MainWindow", "Settings"))

        self.actionNew.setText(QtCore.QCoreApplication.translate("MainWindow", "Open"))
        self.actionNew.setToolTip(QtCore.QCoreApplication.translate("MainWindow", "Load new file"))
        self.actionSettings.setText(QtCore.QCoreApplication.translate("MainWindow", "Settings"))
        self.actionHelp.setText(QtCore.QCoreApplication.translate("MainWindow", "Help"))
        
